persistence/src/test/java/com/epam/rft/atsy/persistence/repositories/CandidateRepositoryIT.java

```diff
@@ -0,0 +1,323 @@
+        }
+    }
+
+    @Test
+    public void findAllCandidatesByFilterRequestWithValidDataSortedByPhoneInDescendingOrder() {
+        for (CandidateEntity candidate : POSITIVE_CANDIDATES_TEST) {
+            //Given
+            final String name = candidate.getName();
+            final String email = candidate.getEmail();
+            final String phone = candidate.getPhone();
+
+            //When
+            List<CandidateEntity> result = this.candidateRepository.findAllCandidatesByFilterRequest(
+                    name,
+                    email,
+                    phone,
+                    new Sort(createSortOrder(Sort.Direction.DESC, "phone")));
+
+            //Then
+            assertCandidateEntityListWithSorting(result, name, email, phone, createSortOrder(Sort.Direction.DESC, "phone"));
+        }
+    }
+
+    @Test
+    public void findAllCandidateByFilterRequestWithInvalidData() {
+        for (CandidateEntity candidate : NEGATIVE_CANDIDATES_TEST) {
+            //Given
+            final String name = candidate.getName();
+            final String email = candidate.getEmail();
+            final String phone = candidate.getPhone();
+
+            //When
+            List<CandidateEntity> result = this.candidateRepository.findAllCandidatesByFilterRequest(
+                    name,
+                    email,
+                    phone,
+                    null);
+
+            //Then
+            assertCandidateEntityListWithInvalidData(result);
+
+        }
+    }
+
+    private void assertCandidateEntityListWithInvalidData(List<CandidateEntity> candidates) {
+        assertThat(candidates, notNullValue());
+        assertThat(candidates, empty());
+    }
+
+    private void assertCandidateEntityListWithSorting(List<CandidateEntity> candidates,
+                                                      String expectedName,
+                                                      String expectedEmail,
+                                                      String expectedPhone,
+                                                      Sort.Order sorting) {
+
+        assertThat(candidates, notNullValue());
+
+        for (CandidateEntity entity : candidates) {
+
+            assertThat(entity, notNullValue());
+
+            if (!expectedName.isEmpty()) {
+                assertThat(entity.getName(), notNullValue());
+                assertThat(entity.getName(), containsString(expectedName));
+            }
+
+            if (!expectedEmail.isEmpty()) {
+                assertThat(entity.getEmail(), notNullValue());
+                assertThat(entity.getEmail(), containsString(expectedEmail));
+            }
+
+            if (!expectedPhone.isEmpty()) {
+                assertThat(entity.getPhone(), notNullValue());
+                assertThat(entity.getPhone(), containsString(expectedPhone));
+            }
+        }
+
+        boolean invariant = true;
+        if (!candidates.isEmpty()) {
```

Comment:
Please rework this bunch of if-s and for-s.

As the test data has only 3 items, you can create named constants for them, also can create predefined lists for all tested sorting, so a single test would look like this:

> assertThat(actualCandidates.size(), is(expectedCandidates.size()));
> for(Iterator<CandidateEntity> expected = expectedCandidates.iterator(), actual = actualCandidates.iterator(); expected.hasNext() && actual.hasNext();) {
> assertThat(actual.next(), is(expected.next()));
> }